### PR TITLE
phraze: init at 0.3.11

### PR DESCRIPTION
**Pull request by x123**
## Description of changes

Adding phraze, a passphrase generation utility written in Rust:

https://github.com/sts10/phraze

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
No, but false, relaxed, and true all work.
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/ph/phraze/package.nix [hunk 1]
```diff
@@ -0,0 +1,27 @@
+{ lib
+, fetchFromGitHub
+, rustPlatform
+,
+}:
```
Comment (AndersonTorres) [inline pkgs/by-name/ph/phraze/package.nix:5]:
```suggestion
{ 
  lib,
  fetchFromGitHub,
  rustPlatform,
}:

```

File: pkgs/by-name/ph/phraze/package.nix [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+rustPlatform.buildRustPackage rec {
+  pname = "phraze";
+  version = "0.3.10";
+
+  src = fetchFromGitHub {
+    owner = "sts10";
+    repo = "phraze";
+    rev = "c88511d6f267076d8724c93d2067ac349957a0d0";
```
Comment (AndersonTorres) [inline pkgs/by-name/ph/phraze/package.nix:13]: Why not to use version here?

File: pkgs/by-name/ph/phraze/package.nix [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+  cargoHash = "sha256-3NmLvDNriXKBPsilK8x0gwzi3krIdVN5Bmi6ib3Dxe4=";
+
+  meta = with lib; {
+    description = "Generate random passphrases";
+    homepage = "https://github.com/sts10/phraze";
+    changelog = "https://github.com/sts10/phraze/releases/tag/v${version}";
+    license = licenses.mpl20;
+    maintainers = with maintainers; [ x123 ];
+    mainProgram = "phraze";
+  };
```
Comment (AndersonTorres) [inline pkgs/by-name/ph/phraze/package.nix:26]:
```suggestion
  meta = {
    description = "Generate random passphrases";
    homepage = "https://github.com/sts10/phraze";
    changelog = "https://github.com/sts10/phraze/releases/tag/v${version}";
    license = lib.licenses.mpl20;
    maintainers = with lib.maintainers; [ x123 ];
    mainProgram = "phraze";
  };
```

**Review by AndersonTorres — CHANGES_REQUESTED** [review]

File: pkgs/by-name/ph/phraze/package.nix (hunk 2 above)
Comment (x123) [reply pkgs/by-name/ph/phraze/package.nix]: @AndersonTorres  Fixed. Somewhere in the docs it mentions a full commit hash being "best", but the tag does make the most sense.

**Comment by AndersonTorres** [comment]
Squash the commits and reword them according to contributing guidelines.

File: pkgs/by-name/ph/phraze/package.nix [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+  pname = "phraze";
+  version = "0.3.10";
+
+  src = fetchFromGitHub {
+    owner = "sts10";
+    repo = "phraze";
+    rev = "v${version}";
+    hash = "sha256-Kg3s5iNQajGFT1+YJFdCvq6UoFJS/YDfhdAOp6xV8w4=";
+  };
+
```
Comment (h7x4) [inline pkgs/by-name/ph/phraze/package.nix:17]: Consider enabling tests, they seem reasonable.

```suggestion

  doCheck = true;

```

File: pkgs/by-name/ph/phraze/package.nix [hunk 5]
```diff
@@ -0,0 +1,28 @@
+{
+  lib,
+  fetchFromGitHub,
+  rustPlatform,
+}:
+
+rustPlatform.buildRustPackage rec {
+  pname = "phraze";
+  version = "0.3.10";
```
Comment (h7x4) [inline pkgs/by-name/ph/phraze/package.nix:9]: New release dropped 2 hours ago if you'd like to combine it in this PR: https://github.com/sts10/phraze/releases/tag/v0.3.11

**Review by h7x4 — APPROVED** [review]
Verified [github id](https://api.github.com/user/5481629) and upstream license, diff seems good.

**Comment by h7x4** [comment]
Feel free to ping me for merge once ofborg is done

**Comment by x123** [comment]
> Feel free to ping me for merge once ofborg is done

I will likely be asleep, will ping in the morning if it’s not merged. I’m not in any rush, so no pressure.

**Comment by kirillrdy** [comment]
how come cargoHash changed ? between force pushes

**Comment by x123** [comment]
> how come cargoHash changed ? between force pushes

I don’t believe I properly changed them after the move to 0.3.11, and ofBorg had some failures on it that I didn’t notice. I believe the cargoHash and source hash should be correct now.

**Comment by sts10** [comment]
Phraze creator here. Very flattered by the idea that it's being packaged here! Apologies for the short amount of time between releases v0.3.10 and v0.3.11 -- I'll try to be more patient going forward. 

Please let me know (a) if there's anything I can do to help this packaging process (though it looks to be close to done?) and/or (b) what installation instructions I might include in the [Phraze README](https://github.com/sts10/phraze/blob/main/readme.markdown) for Nix users (I'm not a Nix user myself).

**Comment by h7x4** [comment]
Built and tested the tool on x86_64 nixos. I think the current cargoHash is correct

**Comment by h7x4** [comment]
@sts10 I don't think you need to worry too much about release speed. We're sitting on entire ecosystems worth of packages, many of which takes a lot more effort to update due to complicated build instructions and a ton of dependencies. I'd say feel free to go at your own pace without worrying about nixpkgs, but I guess it would affect x123 more as the package maintainer and the users waiting for updates more than it affects me, so prefer their opinion over mine.

The installation instructions are a bit complicated. In contrast to lots of other package managers, nix doesn't really promote installing stuff in the global environment (with the command `nix-env -iA <package-name>`). It's usually recommended to use the packages through either a nixos/darwin/home-manager configuration, or through a nix-shell (or similar nix-based development environments). However, most nix users are already aware of this, so they will probably install it as they see fit and ignore the command. Maybe it would be best to link to the package through a repology badge or something similar?

**Comment by x123** [comment]
@sts10 no worries about release speed, as this particular package is quite simple and will likely just need the version and hashes updated on each new release. I can also help in the coming days with a pull request to your README with very simple nix usage instructions.

**Pull request merged**